camel/personas/persona_generator.py

```diff
@@ -0,0 +1,217 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+import re
+from typing import List, Optional, Union
+
+from camel.agents import ChatAgent
+from camel.messages.base import BaseMessage
+from camel.models.base_model import BaseModelBackend
+from camel.personas.persona import Persona
+from camel.prompts import TextPrompt
+from camel.types import RoleType
+
+
+class PersonaGenerator(ChatAgent):
+    r"""A generator of personas. This class manages a collection of Persona
+    instances and provides methods for adding, removing, and manipulating
+    personas within the group.
+
+    Args:
+        model (BaseModelBackend, optional): The model to use for persona
+        group_name (str, optional): The name of the group.
+        group_description (str, optional): A description of the group.
+    """
+
+    def __init__(
+        self,
+        model: Optional[BaseModelBackend] = None,
+    ):
+        system_message = BaseMessage(
+            role_name="",
+            role_type=RoleType.ASSISTANT,
+            meta_dict=None,
+            content="",
+        )
+        self.model = model if model else None
+        super().__init__(system_message, model=model)
+        self.personas: List[Persona] = []
+
+    def add_persona(self, persona: Persona):
+        r"""Add a persona to the group."""
+        self.personas.append(persona)
+
+    def __delitem__(self, index: int):
+        r"""Remove a persona from the group by index.
+
+        Args:
+            index (int): The index of the persona to remove.
+        """
+        if 0 <= index < len(self.personas):
```

Comment:
I feel using index is not user friendly, how about add id to `Persona` so that user can remove by using id, or we can remove it from name?

Reply:
I think I have to create an unique id to identify them. I will do this.